hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -163,7 +163,10 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
                 // converted into the same EVM address as the alias
                 final var payerEvmAddress = extractEvmAddress(payerKey);
                 final var accountKeyEvmAddress = extractEvmAddress(op.keyOrThrow());
-                if (!alias.equals(payerEvmAddress) && !alias.equals(accountKeyEvmAddress)) {
+                // payerEvmAddress will be null for cases where we have synthetic crypto-create used to create a
+                // contract as part of a contract call or contract create,
+                // those cases should not be treated as hollow account creation
+                if (payerEvmAddress != null && !alias.equals(payerEvmAddress) && !alias.equals(accountKeyEvmAddress)) {
```

Comment:
@tinker-michaelj Not sure if this is ok to be done here, but was not able to think of an alternative. Wdyt? 🤔

Reply:
Actually if the payer key for the top-level contract call/create is ECDSA key then we'll still end up inside the `if` which does not seem right, so probably we need to do something else. 😞